### PR TITLE
[Kyuubi #4222] Use hiveTableCatalog to updateTableStats instead of sessionCatalog

### LINKED ISSUE #4222: [Bug] Use the hive connector plugin to realize the federated query of multiple hive metastores, and found that the execution of the INSERT INTO statement reported an error
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

Configure according to the following documents
https://kyuubi.readthedocs.io/en/master/connector/spark/hive.html#kyuubi-hive-conf

image

execute select successfully

image
execute insert into，then report an error

image

image

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [X] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #4222 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by jiaoqingbo** [comment]
before modification
image
image
after modification
image

**Review by pan3793 — APPROVED** [review]

File: extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveBatchWrite.scala [hunk 1]
```diff
@@ -67,9 +68,23 @@ class HiveBatchWrite(
     }
 
     val sparkSession = SparkSession.active
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveBatchWrite.scala:70]: Can we pass it from the constructor?

**Comment by pan3793** [comment]
Thanks, merged to master

**Comment by Yikf** [comment]
LGTM from me, sorry for late reply since i'm so busy this week🙁

**Pull request closed without merging**